lib/widgets/read_receipts.dart

```diff
@@ -0,0 +1,208 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+
+import '../api/route/messages.dart';
+import '../generated/l10n/zulip_localizations.dart';
+import 'action_sheet.dart';
+import 'actions.dart';
+import 'color.dart';
+import 'inset_shadow.dart';
+import 'profile.dart';
+import 'store.dart';
+import 'text.dart';
+import 'theme.dart';
+import 'user.dart';
+
+/// Opens a bottom sheet showing who has read the message.
+void showReadReceiptsSheet(BuildContext pageContext, {required int messageId}) {
+  final accountId = PerAccountStoreWidget.accountIdOf(pageContext);
+
+  showModalBottomSheet<void>(
+    context: pageContext,
+    // Clip.hardEdge looks bad; Clip.antiAliasWithSaveLayer looks pixel-perfect
+    // on my iPhone 13 Pro but is marked as "much slower":
+    //   https://api.flutter.dev/flutter/dart-ui/Clip.html
+    clipBehavior: Clip.antiAlias,
+    useSafeArea: true,
+    isScrollControlled: true,
+    builder: (_) {
+      return PerAccountStoreWidget(
+        accountId: accountId,
+        child: SafeArea(
+          minimum: const EdgeInsets.only(bottom: 16),
+          child: ReadReceipts(messageId: messageId)));
+    });
+}
+
+/// The read receipts sheet.
+///
+/// Figma link:
+///   https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=11367-20647&t=lSnHudU6l7NWx0Fa-0
+class ReadReceipts extends StatefulWidget {
+  const ReadReceipts({super.key, required this.messageId});
+
+  final int messageId;
+
+  static const _helpCenterReference = '/help/read-receipts';
```

Comment:
nit: "reference" is a bit vague about what sort of thing this is.

I think "relative URL" is a good term. The official term in the spec would be "relative-URL string":
https://url.spec.whatwg.org/#relative-url-string

and here the type says it's a string so we don't need to repeat that part. Therefore:

```suggestion
  static const _helpCenterRelativeUrl = '/help/read-receipts';
```

(In the now-obsolete RFCs that previously defined the concept of a URL, [this was](https://www.rfc-editor.org/rfc/rfc3986#section-4) a "URI-reference" and more specifically a "relative reference". Perhaps that's what you had in mind with this name.)

Reply:
separate nit: this private const can go on the one class that uses it, and then the reference to it doesn't need to repeat the class name.